src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -136,7 +136,17 @@ def create(self, cluster_resource: Cluster) -> Simplified:
                 json.dumps(cluster_resource)).to_simplified()
         except RuntimeError as e:
             raise DashboardException(e, component='smb')
-
+    
+    @DeletePermission
+    @EndpointDoc("Remove smb cluster",
+                 parameters={
+                     'cluster_id': (str, 'Unique identifier for the cluster')},
+                  responses={204: None})
+    def delete(self,cluster_id: str): 
+      """
+        Remove an smb cluster 
+      """
+      return mgr.remote('smb', 'cluster_rm', cluster_id).to_simplified()
```

Comment:
Instead of invoking `cluster_rm` I think we should follow the other endpoint's approach and call `apply_resources` and apply the cluster resource with `intent = removed` to remove it. This way, I believe, we can avoid possibly side effects that `cluster_rm` might do.

Also, while checking this, I realized of a mistake. When deleting a resource goes unsucessfully we are not properly handeling the error, and this would cause that we would recieve a positive notification even though the cluster was not being removed, as can be seen in the following video:

[screen-capture (16).webm](https://github.com/user-attachments/assets/d9a72eda-5034-4141-9744-07efb80ad50e)

Here, in the video, I'm trying to delete a cluster that has a share, which should not be possible, hence the cluster does not get deleted but we are getting a notification as if it successfully deleted it.

To fix this I've raised this PR #61123 that introduces a decorator to handle this. It raises a DashboardException displaying the msg when a resource update is unsucessful.
Applying this decorator we can see that a notification is now displayed stating the error message:

[screen-capture (17).webm](https://github.com/user-attachments/assets/c7018ed7-ee61-4b1b-9d57-d016847e5a12)